app/views/exchanges/employer_applications/index.js.erb

```diff
@@ -0,0 +1,223 @@
+var containing_row = $("#<%= @element_to_replace_id %>").closest('TR');
+var $parent_row = containing_row;
+//var $parent_row = $(".table:first > tbody > tr:not(tr.child-row)").eq(<%= @row %>);
+if ( $('tr.child-row:visible').length > 0 ) {
+  $('tr.child-row:visible').remove();
+  $("li>a:contains('Collapse Form')").addClass('disabled');
+
+}
+
+var actionSelected = "";
+var applicationId = "";
+var employerId = "";
+var pyEndDate = "";
+var termReason = "";
+
+if ( $parent_row.next().hasClass('child-row') ) {
+//
+} 
+else {
+  $parent_row.after('<tr class="child-row"><td colspan="100"><%= j render "index"%></td></tr>');
+  $("li>a:contains('Collapse Form')").eq(containing_row.index()).removeClass('disabled');
+  $('.dropdown.pull-right').removeClass('open');
+}
+
+$('input[name=plan_year_radio]').click(function() {
+  hidePlanYearActions();
+  clearDatePickerValue();
+  hideTerminationRadios();
+  $('.py-btn-grp').addClass('disabled');
+  var pyStatus = $(this).attr("status");
+  populateDropDown(pyStatus);
+  $('input[name=plan_year_radio]:checked').closest('td').siblings().find('.py-btn-grp').removeClass('disabled')
+});
+
+$('#plan-year-btn-group ul li').each(function() {
+  $(this).click(function() {
+    // Do Something
+  })
+});
+
+function populateDropDown(status) {
```

Comment:
Currently term/cancel buttons being visible in the UI but throwing an error when submitted at term/cancel, due to model level missing state transitions.

Please modify method as below....to match possible state transitions as per backend

function populateDropDown(status) {

  if cancel_condition
    $('li.cancel-py').removeClass('blocking');
  else
    $('li.cancel-py').addClass('blocking');
  end 

  if term_condition
    $('li.terminate-py').removeClass('blocking');
  else
    $('li.terminate-py').removeClass('blocking');
  end